submarine-server/server-core/src/main/java/org/apache/submarine/server/job/JobManager.java

```diff
@@ -0,0 +1,214 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.server.job;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.ws.rs.core.Response.Status;
+
+import org.apache.submarine.commons.utils.SubmarineConfiguration;
+import org.apache.submarine.server.SubmarineServer;
+import org.apache.submarine.server.SubmitterManager;
+import org.apache.submarine.server.api.exception.SubmitterException;
+import org.apache.submarine.server.api.job.JobSubmitter;
+import org.apache.submarine.server.api.job.Job;
+import org.apache.submarine.server.api.job.JobId;
+import org.apache.submarine.server.api.spec.JobSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * It responsible for manage the job (CRUD) and cached the job.
+ */
+public class JobManager {
+  private static final Logger LOG = LoggerFactory.getLogger(JobManager.class);
+
+  private static volatile JobManager manager;
+
+  private final AtomicInteger jobCounter = new AtomicInteger(0);
+
+  /**
+   * Used to cache the specs by the job id.
+   *  key: the string of job id
+   *  value: Job object
+   */
+  private final ConcurrentMap<String, Job> cachedJobMap = new ConcurrentHashMap<>();
+
+  private SubmitterManager submitterManager;
+
+  /**
+   * Get the singleton instance
+   * @return object
+   */
+  public static JobManager getInstance() {
+    if (manager == null) {
+      synchronized (JobManager.class) {
+        if (manager == null) {
+          SubmarineConfiguration conf = SubmarineConfiguration.getInstance();
+          SubmitterManager submitterManager = new SubmitterManager(conf);
+          manager = new JobManager(submitterManager);
+        }
+      }
+    }
+    return manager;
+  }
+
+  private JobManager(SubmitterManager submitterManager) {
+    this.submitterManager = submitterManager;
+  }
+
+  /**
+   * Create job
+   * @param spec job spec
+   * @return object
+   * @throws JobServiceException the service error
+   */
+  public Job createJob(JobSpec spec) throws JobServiceException {
+    try {
+      JobSubmitter submitter = getSubmitter(spec.getSubmitterSpec().getType());
+      Job job = submitter.createJob(spec);
+      job.setJobId(generateJobId());
+      job.setSpec(spec);
+      cachedJobMap.putIfAbsent(job.getJobId().toString(), job);
+      return job;
+    } catch (SubmitterException e) {
+      throw JobServiceException.fromSubmitterException(e);
```

Comment:
Now the extended `*Exception` makes the system very complicated.
It is sufficient to use `SubmarineRuntimeException` in the commons-utils module.